code/game/objects/items/devices/spyglasses.dm

```diff
@@ -38,58 +38,41 @@
 	return ..()
 
 
-/obj/item/clothing/accessory/spy_bug
-	name = "pocket protector"
-	icon = 'icons/obj/clothing/accessories.dmi'
-	icon_state = "pocketprotector"
-	desc = "An advanced piece of espionage equipment in the shape of a pocket protector. It has a built in 360 degree camera for all your \"admirable\" needs. Microphone not included."
+/obj/item/pen/spy_bug
+	desc = "An advanced piece of espionage equipment in the shape of a pen. It has a built in 360 degree camera for all your \"admirable\" needs. Microphone not included."
 	var/obj/item/clothing/glasses/sunglasses/spy/linked_glasses
 	var/atom/movable/screen/map_view/cam_screen
 	var/list/cam_plane_masters
 	// Ranges higher than one can be used to see through walls.
 	var/cam_range = 1
 	var/datum/movement_detector/tracker
 
-/obj/item/clothing/accessory/spy_bug/Initialize()
-	. = ..()
-	tracker = new /datum/movement_detector(src, CALLBACK(src, .proc/update_view))
-
-	cam_screen = new
-	cam_screen.name = "screen"
-	cam_screen.assigned_map = "spypopup_map"
-	cam_screen.del_on_map_removal = FALSE
-	cam_screen.set_position(1, 1)
-
-	// We need to add planesmasters to the popup, otherwise
-	// blending fucks up massively. Any planesmaster on the main screen does
-	// NOT apply to map popups. If there's ever a way to make planesmasters
-	// omnipresent, then this wouldn't be needed.
-	cam_plane_masters = list()
-	for(var/plane in subtypesof(/atom/movable/screen/plane_master))
-		var/atom/movable/screen/instance = new plane()
-		instance.assigned_map = "spypopup_map"
-		instance.del_on_map_removal = FALSE
-		instance.screen_loc = "spypopup_map:CENTER"
-		cam_plane_masters += instance
-
-/obj/item/clothing/accessory/spy_bug/Destroy()
+/obj/item/pen/spy_bug/ComponentInitialize()
+	..()
+	AddComponent(/datum/component/spybug)
+
+/obj/item/pen/spy_bug/Destroy()
 	if(linked_glasses)
 		linked_glasses.linked_bug = null
 	qdel(cam_screen)
 	QDEL_LIST(cam_plane_masters)
 	qdel(tracker)
 	return ..()
 
-/obj/item/clothing/accessory/spy_bug/proc/update_view()//this doesn't do anything too crazy, just updates the vis_contents of its screen obj
-	cam_screen.vis_contents.Cut()
-	for(var/turf/visible_turf in view(1,get_turf(src)))//fuck you usr
-		cam_screen.vis_contents += visible_turf
-
 //it needs to be linked, hence a kit.
 /obj/item/storage/box/rxglasses/spyglasskit
 	name = "spyglass kit"
 	desc = "this box contains <i>cool</i> nerd glasses; with built-in displays to view a linked camera."
 
+/obj/item/storage/box/rxglasses/spyglasskit/PopulateContents()
+	var/obj/item/clothing/accessory/pocketprotector/protector = new (src)
+	var/obj/item/pen/spy_bug/newbug = new(protector)
+	var/datum/component/spybug/spy_bug_component = newbug.GetComponent(/datum/component/spybug)
```

Comment:
isn't this added on initialize

Reply:
The contents? Or the component?

This part of the code is used to connect the spyglasses to the spybug item.